engine/context/src/main/java/io/deephaven/engine/context/StandaloneQueryScope.java

```diff
@@ -0,0 +1,153 @@
+        if (value instanceof String) {
+            final String stringValue = (String) value;
+
+            if (!stringValue.isEmpty() && stringValue.charAt(0) == '\''
+                    && stringValue.charAt(stringValue.length() - 1) == '\'') {
+                final String datetimeString = stringValue.substring(1, stringValue.length() - 1);
+
+                final Instant instant = DateTimeUtils.parseInstantQuiet(datetimeString);
+                if (instant != null) {
+                    return instant;
+                }
+
+                final long localTime = DateTimeUtils.parseDurationNanosQuiet(datetimeString);
+                if (localTime != QueryConstants.NULL_LONG) {
+                    return localTime;
+                }
+
+                final Period period = DateTimeUtils.parsePeriodQuiet(datetimeString);
+                if (period != null) {
+                    return period;
+                }
+
+                final Duration duration = DateTimeUtils.parseDurationQuiet(datetimeString);
+                if (duration != null) {
+                    return duration;
+                }
+
+                throw new RuntimeException("Cannot parse datetime/time/period : " + stringValue);
+            }
+        }
+
+        return value;
+    }
+
+    @Override
+    public Set<String> getParamNames() {
+        return valueRetrievers.keySet();
+    }
+
+    @Override
+    public boolean hasParamName(String name) {
+        return valueRetrievers.containsKey(name);
+    }
+
+    @Override
+    public <T> QueryScopeParam<T> createParam(final String name) throws MissingVariableException {
+        final ValueRetriever<?> valueRetriever = valueRetrievers.get(name);
+        if (valueRetriever == null) {
+            throw new MissingVariableException("Missing variable " + name);
+        }
+        // noinspection unchecked
+        return (QueryScopeParam<T>) valueRetriever.createParam();
+    }
+
+    @Override
+    public <T> T readParamValue(final String name) throws MissingVariableException {
+        final ValueRetriever<?> valueRetriever = valueRetrievers.get(name);
+        if (valueRetriever == null) {
+            throw new MissingVariableException("Missing variable " + name);
+        }
+        // noinspection unchecked
+        return (T) valueRetriever.getValue();
+    }
+
+    @Override
+    public <T> T readParamValue(final String name, final T defaultValue) {
+        final ValueRetriever<?> valueRetriever = valueRetrievers.get(name);
+        if (valueRetriever == null) {
+            return defaultValue;
+        }
+        // noinspection unchecked
+        return (T) valueRetriever.getValue();
+    }
+
+    @Override
+    public <T> void putParam(final String name, final T value) {
+        if (value instanceof LivenessReferent && DynamicNode.notDynamicOrIsRefreshing(value)) {
+            manage((LivenessReferent) value);
+        }
+        NameValidator.validateQueryParameterName(name);
```

Comment:
Should validate before managing...
```suggestion
        NameValidator.validateQueryParameterName(name);
        if (value instanceof LivenessReferent && DynamicNode.notDynamicOrIsRefreshing(value)) {
            manage((LivenessReferent) value);
        }
```